src/YoutubeAtom.tsx

```diff
@@ -76,11 +248,117 @@ export const YoutubeAtom = ({
     title = 'YouTube video player',
     duration,
     origin,
+    gaEventEmitter,
+    ophanEventEmitter,
 }: YoutubeAtomType): JSX.Element => {
     const embedConfig =
         adTargeting && JSON.stringify(buildEmbedConfig(adTargeting));
     const originString = origin ? `&origin=${origin}` : '';
     const iframeSrc = `https://www.youtube.com/embed/${videoMeta.assetId}?embed_config=${embedConfig}&enablejsapi=1${originString}&widgetid=1&modestbranding=1`;
+
+    const [hasUserLaunchedPlay, setHasUserLaunchedPlay] = useState<boolean>(
+        false,
+    );
+    const [isPlayerReady, setIsPlayerReady] = useState<boolean>(false);
+    const [player, setPlayer] = useState<YT.Player | null>(null);
+
+    const loadVideo = () => {
+        setPlayer(
+            new window.YT.Player(`${videoMeta.assetId}`, {
+                videoId: `${videoMeta.assetId}`,
+                events: {
+                    onReady: () => setIsPlayerReady(true),
+                    // Issue with setting events on Youtube object (refer to useEffect below)
+                    // https://stackoverflow.com/a/17078152
+                    // onPlayerStateChange,
+                },
+                playerVars: {
+                    mute: 1,
+                    autoplay: 1,
+                },
+            }),
+        );
+    };
+
+    useEffect(() => {
+        if (player) {
+            // ts-ignore is used because onStateChange has different listener to what is actually sent
+            // @ts-ignore
+            const playerAnalyticsFunction = (e) =>
+                onPlayerStateChangeAnalytics({
+                    e,
+                    hasUserLaunchedPlay,
+                    setHasUserLaunchedPlay,
+                    gaEventEmitter,
+                    ophanEventEmitter,
+                    getCurrentTime: player.getCurrentTime,
+                    getDuration: player.getDuration,
+                });
+            // Issue with setting events on Youtube object
+            // https://stackoverflow.com/a/17078152
+
+            player.addEventListener('onStateChange', playerAnalyticsFunction);
+        }
+    }, [
+        player,
+        hasUserLaunchedPlay,
+        setHasUserLaunchedPlay,
+        gaEventEmitter,
+        ophanEventEmitter,
+    ]);
+
+    useEffect(() => {
+        // if window is undefined it is because this logic is running on the server side
+        if (typeof window === 'undefined') return;
+
+        if (window.YT) {
+            loadVideo();
+        } else {
+            // If not, load the script asynchronously
+            const tag = document.createElement('script');
+            tag.src = 'https://www.youtube.com/iframe_api';
```

Comment:
We have a `loadScript` util in libs https://github.com/guardian/libs/blob/main/src/loadScript.ts

Reply:
I tried using this lib but got an error in storybook
```
ERROR in ./node_modules/@guardian/libs/dist/esm/loadScript.js 25:19
Module parse failed: Unexpected token (25:19)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|     };
|     const ref = document.scripts[0];
>     ref.parentNode?.insertBefore(script, ref);
| });
```
I am unsure the cause, I suspect it could be something to do with typescript, webpack and storybook (suspect it doesn't understand `?`)

Reply:
@amyhughes  and I saw a similar issue when trying to use this function. @sndrs off the top of your head is this a problem our end or in the `/libs` project?

Reply:
Actually, ignore me. This is caused by us using an older version of Node